### PR TITLE
Project: Backgrounds and Gradients 

### PR DESCRIPTION
**Pull request by rkyankya**
Microverse
Apple Clone 
Project using HTML and CSS

### CODE REVIEW

**Comment by samgaco** [comment]
Hi @rkyankya you have submitted this project as **Embedding Images and Video** , but is **Building with Backgrounds and Gradients** , so please submit back for another code review for the right project. Regards :hand:

File: README.md
Comment (samgaco) [inline README.md:16]: The readme file styling could be improved further, find some inspiration [here](https://github.com/othneildrew/Best-README-Template).

**Review by samgaco — CHANGES_REQUESTED** [review]
Hi @rkyankya :high_brightness: 

1. The search bar is not appearing in the navigation bar as it should (at the rightmost side of it). You can confirm it in the [live-demo.](https://rawcdn.githack.com/rkyankya/Apple_Clone/1b4c8fc565eb778e94a5e757d62147c506cd81a7/index.html)

2. Read about [HTML semantic](https://www.w3schools.com/html/html5_semantic_elements.asp) elements and incorporate it into your project. ( Use `<main>` , `<aside>` , `<section>` , `<article>` when necessary )

3. Stickler is not active on this repository yet. You have more information about how to implement it [here](https://github.com/microverseinc/linters-config/tree/master/css). Once you have it working, you should correct the errors if they come up.

4. Work on the size of the four boxes below, right now they are not occupying all the space of the site as in the original version.
---------------------

moving on :arrow_down: 
* **[For your next code review requests]**  Add a short summary of changes in Pull Request description

-----------------------
Please :pray: work on these 4 points and submit back for another code review. The comments below are optional (about the readme file) 

* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

File: README.md
Comment (bolah2009) [inline README.md:11]: Let's make the `README` file more descriptive, details like what was done in the projects, what was learned, link to the assignment, clone and source page, screenshots of the clone page can make README look more professional.
 > Remember about descriptive README file.

File: stylelint.config.js
Comment (bolah2009) [inline stylelint.config.js]: #### `AWESOME`

Awesome job adding `stylelint` config file. :+1: 
However, you forgot to add the `.stickler.yml` file? :thinking: 

----

#### `REQUIRED`

Add `.stickler.yml` file to the root of the project. Also let's use [this file](https://github.com/microverseinc/linters-config/blob/master/css/.stickler.yml). :point_down: 

``` yml
linters:
  stylelint:
    config: 'stylelint.config.js'
```

If you are having an issue with setting up linter, read the `Troubleshooting` section of [this repo](https://github.com/microverseinc/linters-config/tree/master/css).
Remember to:
- Install stickler-ci https://github.com/apps/stickler-ci
- Enable stickler in your repo by turning this repo on on stickler website [here](https://stickler-ci.com/).

After setting up the linter and fixing all possible errors this PR (Pull Request) should have a section like this at the end of the page. :point_down: 
  succesful check PR

File: index.html
Comment (bolah2009) [inline index.html]: Let's use `footer` tag here instead

File: index.html
Comment (bolah2009) [inline index.html:77]: Let's wrap this element with the same `footer` element

File: index.html
Comment (bolah2009) [inline index.html:38]: Consider using a `main` tag here instead

File: index.html
Comment (bolah2009) [inline index.html]: Why not use a `nav` element here instead?

File: assets/css/main.css
Comment (bolah2009) [inline assets/css/main.css]: #### `REQUIRED`

Always look for a way to use [specificity](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity) before even considering `!important`. 

#### In this case, move this property after the `background` property, this will ensure that the property is not overridden by the `backgroud` shorthand property.

```suggestion
    background: url("../image/search-icon-black.png") top 3px left 4px no-repeat;
    background-color: white !important;
```

Learn more [here](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity#The_!important_exception)

> Make sure you DON’T make [common mistakes](https://speckyboy.com/good-bad-css-practices/) in your CSS styles.

> **Recommendation:** You should only use the !important rule when overriding styles in a user style sheet, overriding 3rd party code and inline styles as well as utility classes.

File: assets/css/main.css
Comment (bolah2009) [inline assets/css/main.css]: Please, let's remove this too.

File: assets/css/main.css
Comment (bolah2009) [inline assets/css/main.css]: Why do we have a margin here? :thinking: 

Currently, the margin is affecting hover effects on the nav links. See below:

 hover apple clone


#### Please consider removing it.

File: assets/css/main.css
Comment (bolah2009) [inline assets/css/main.css:149]: Is there a reason why this is having this property `min-width: 1024px;`?
Currently, the footer is not aligned with the main content, see illustration below. :point_down: 

apple footer not aligned

#### Please, consider deleting this property.

**Review by bolah2009 — CHANGES_REQUESTED** [review]
# Nice job @rkyankya  :smile: 

### The [clone](https://raw.githack.com/rkyankya/Apple_Clone/fb4829a870ebe5a2f48809148012dcfc2d8f78ae/index.html) looks good to me, uses CSS Gradients and Backgrounds effectively. :clap: 

However, there are a few changes required to be made before a :+1: 
- Setup linter and fix possible errors/warnings. [See comments below](https://github.com/rkyankya/Apple_Clone/pull/1#discussion_r326869813)
- Make `README` file more descriptive
  > Remember about descriptive README file.
- Let's make this Pull Request look more professional. [See below](https://github.com/rkyankya/Apple_Clone/pull/1#pullrequestreview-291476783)
  - [NEXT PROJECT]: Use meaningful commits messages. [Learn more here](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8/)
  - [REQUIRED] Use meaningful names for  Pull Requests.
  - [REQUIRED] Add a short summary of changes in Pull Request description.
- [NEXT PROJECTS]: For next (HTML/CSS) project, please follow `GitHub flow` [guidelines](https://microverse.pathwright.com/library/microverse-onboarding/86777/path/step/53987922/). What is missing here is a descriptive branch name instead of `Review`, think of feature branch as a branch name that is descriptive enough to have an idea of what the branch is used for.

#### :recycle:  Make these changes and submit another code review request. :+1: 

#### Feel free to reach me on slack if you having issues implementing these features. :+1: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

**Review by bolah2009 — COMMENTED** [review]
# Rename PR

  rename PR

-----

# Update PR description

  update-PR-description-html

File: README.md
Comment (whiz25) [inline README.md]: Please make this README more descriptive. This will help other developers understand how to get started with the project. It is required for this project's completion. Look at this [post](https://medium.com/@meakaakka/a-beginners-guide-to-writing-a-kickass-readme-7ac01da88ab3)

**Review by whiz25 — CHANGES_REQUESTED** [review]
Fix the conflict in ```index.html``` and send for another review.

**Review by SunnyAsar — CHANGES_REQUESTED** [review]
## Hello @rkyankya 
well done on the project, it is looking good.
#### A lot of the changes requested by @bolah2009  are still not fixed.
- get the margins removed for the nav, it doesn't look right on hover.
- use the right html semantics like `footer` `nav` and `main`
- avoid using `!import` in css
- please setup you linter  for the repo as well

### Basically let's get all the previous requirements fixed.

**Request another review when its ready** Cheers!

**Comment by Torres-ssf** [comment]
### Pull Request Under Review By Sérgio (TSE)

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
### Hi there @rkyankya :hand: 
### You did a nice job here, your page looks great :tada: :clap: 

However, as was said by the previous reviewers, we still need to add the .stickler.yml file to your project. Linting is important to reduce errors and improve the overall quality of your code. Using the linter also gives a professional look to your repository. It's so important that we TSEs are not allowed to approve projects when the linter was not properly set or with errors not solved. I'm really sorry about this and I know that can be frustrating, but this change will add high value to your repo as well as your portfolio.

- [ ] Kindly add the .stickler.yml file to your repo. You may also need to fix any errors if it found any. Please follow the instructions [here](https://github.com/microverseinc/linters-config/tree/master/css).

Your project approval is very close :smiley: 
Happy coding :computer:

File: README.md
Comment (samgaco) [inline README.md:9]: The presentation of the readme file is really important since it's the first thing people will see when checking your projects, and it's a good skill to build along the Microverse journey, the more you work on that, the better you will be at it and with more ease you will be able to do great readme files that make you look like a serious professional. So, please, haven into account [this template](https://github.com/othneildrew/Best-README-Template) and try to add some changes in this project or moving on.

**Review by samgaco — CHANGES_REQUESTED** [review]
Hi @rkyankya all the previous requirements have been met hence the project should be approved, but there are two things to address before:

* [**REQUIERED]** You should solve the conflicts before merging.
* **[REQUIERED]** There is a[ horizontal scrolling bar showing up in](https://rawcdn.githack.com/rkyankya/Apple_Clone/fa6ed3c8de41dc2a85d4cdab549729287249d419/index.html) your project, please find the cause and correct it.

--------------------

**OPTIONAL:** 
* **[OPTIONAL]** Try to improve the style of the readme file

-----------------
* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

**Pull request closed without merging**
